src/components/NetworkMap.vue

```diff
@@ -3,71 +3,137 @@
+      // h and w are full height and width of the Robinson map without any cropping - the correct ratio is 1.97165551906973
+      const w = mapWidth
+      const h = w / 1.97165551906973
 
-      const x = (lng - this.mapLngLeft) * (this.mapWidth / mapLngDelta)
-      const y = this.mapHeight - ((worldMapWidth / 2 * Math.log((1 + Math.sin(latitudeRad)) / (1 - Math.sin(latitudeRad)))) - mapOffsetY)
+      const { X, Y } = this.getXYCoefficients(lat)
+
+      // R = radius of the globe at scale of map
+      const R = mapWidth / 2 / Math.PI / 0.8487
+
+      // longtitude of central point of map in radians
+      const lngRad = (lng + this.lngOffset) * Math.PI / 180
+
+      // x and y co-ordinates ((0,0) at the top-left corner / (mapWidth,mapHeight) at bottom right corner)
+      // 1.02 multiplier of x value is a "fudge" value as the points don't quite match up
+      const x = (w / 2) + (0.8487 * R * X * lngRad * 1.02) + leftOffset
+      const y = (h / 2) - (1.3523 * R * Y) - topOffset
 
       return { x, y }
     },
-    async drawPoints() {
-      const canvas = this.$refs.mapCanvas
-      const ctx = canvas.getContext('2d')
-      ctx.canvas.width = this.mapWidth
-      ctx.canvas.height = this.mapHeight
-      ctx.fillStyle = '#0ecc5f'
-      ctx.strokeStyle = '#5cbd64'
+    getXYCoefficients(lat) {
+      const absoluteLat = Math.abs(lat)
+      // determine if lattitude is positive or negative
+      const sign = lat < 0 ? -1 : 1
 
+      // disallow invalid latitude
+      if (absoluteLat > 90) return
+
+      // if x is multiple of 5, return known X and Y values
+      const i = latArray.indexOf(absoluteLat)
+      if (i >= 0) {
+        const X =  XValues[i]
+        let Y = YValues[i] * sign
+        return { X, Y }
+      }
+
+      // if x is not multiple of 5, perform linear interpolation between two closest known points
+      const lowerLat = Math.floor(absoluteLat / 5) * 5
+      const upperLat = lowerLat + 5
+      const iLower = lowerLat / 5
+      const iUpper = iLower + 1
+      const X = this.linearInterpolation(absoluteLat, lowerLat, upperLat, XValues[iLower], XValues[iUpper])
+      let Y = this.linearInterpolation(absoluteLat, lowerLat, upperLat, YValues[iLower], YValues[iUpper]) * sign
+      return { X, Y }
+    },
+    linearInterpolation(x, x0, x1, y0, y1) {
+      return y0 + ((y1 - y0) * ((x - x0) / (x1 - x0)))
+    },
+    updateMarkers() {
+      // height and width of the map image in the browser
+      const mapWidth = this.$refs.mapImage.clientWidth
+      const mapHeight = this.$refs.mapImage.clientHeight
+      // radius of the marker - set at 5px when map at full width (at 1180px width)
+      const markerRadius = 5 / 1180 * mapWidth
+      // topOffset is number of pixels cropped from top of map when map, calculated as being 5 pixels when map at full height of 515
+      const topOffset = 5 / 515 * mapHeight
```

Comment:
More magic numbers for original map dimensions and related measurements. Please define as named constants.